test/helpers/position_assertions.h

```diff
@@ -305,33 +306,44 @@ class ApplyCodeActionAssertion final : public RangeAssertion {
     std::string toString() const override;
 };
 
-// # symbol-search: "query"
+// # ^^^ symbol-search: "query" [, optional_key = value ]*
```

Comment:
It is a `RangeAssertion`, so yeah, it accepts/knows the range, but it doesn't (currently) do anything interesting with the columns (it only checks the line number).  Here are two ideas re: what we could decide to do with it...

1) Add an optional `range` parameter to the check, perhaps with values like `exact|contains|intersects|ignore` so that we can confirm the range for lines with multiple symbols:
```
attr_writer :foo, :bar, :baz
#                  ^^^ symbol-search: "ba", range="contains"
#                        ^^^ symbol-search: "ba", range="contains"
```

2) . Use the range to extract a default value for the `name` check:
```
def foo_bar(x, y, z)
#   ^^^^^^^ symbol-search: "foba"  (==> range implies expected name="foo_bar")
```

Reply:
Interesting. I'm a little skeptical, considering that it seems like for the most part `workspace/symbol` is purely a function of text to list of symbols (with no reference to a particular textDocument). We have other ways of asserting that the contents of GlobalState is correct; the `workspace/symbol` tests should test specifically whether the additional logic to implement this LSP method is also correct.